src/NuGet.Clients/NuGet.VisualStudio.Common/IDE/VsHierarchyUtility.cs

```diff
@@ -71,7 +71,12 @@ public static bool HasUnsupportedProjectCapability(IVsHierarchy hierarchy)
             return hierarchy.IsCapabilityMatch(ProjectCapabilities.SharedAssetsProject);
         }
 
-        public static string[] GetProjectTypeGuids(IVsHierarchy hierarchy, string defaultType = "")
+        /// <summary>
+        /// Gets the project type guids from the hierarchy.
+        /// </summary>
+        /// <param name="hierarchy">hierarchy to use</param>
```

Comment:
```suggestion
        /// <param name="hierarchy">hierarchy for a project</param>
```

IVsHierarchy represents any node in the Solution Explorer, such as the solution root node, solution folders, files, nodes representing physical folders, virtual nodes like "references" or "dependencies", etc. I suspect trying to get `VSHPROPID_TypeGuid` on most of these node types that are not a project will fail.